### PR TITLE
Check No API Breakages

### LINKED ISSUE #50: Add CI step to catch breaking API changes
It's probably worthwhile adding in CI to catch breaking changes at this stage. Not sure how CI is setup for this repo though.

_Originally posted by @adam-fowler in https://github.com/swift-server/webauthn-swift/issues/43#issuecomment-1952632887_

### PR DESCRIPTION
**Pull request by dimitribouniol**
Added a script to check for breaking API in new PRs, but can't figure out for the life of me where the caller of this script should be. Perhaps it lives in the ci bot configuration directly?

Fixes #50

### CODE REVIEW

**Comment by yim-lee** [comment]
I will take a look.

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Comment by yim-lee** [comment]
A new "api breakage" validation pipeline is set up. The failure is the following:

```
+ docker-compose -f docker/docker-compose.yaml -f docker/docker-compose.2204.59.yaml -p webauthn-api-breakage-prb run --rm shell -cl 'scripts/check_no_api_breakages.sh https://github.com/swift-server/webauthn-swift 12b314f318d15031041fc9a31ae7ecfadada954d main'
Creating network "webauthn-api-breakage-prb_default" with the default driver
Creating webauthn-api-breakage-prb_runtime-setup_1 ... 
Creating webauthn-api-breakage-prb_runtime-setup_1 ... done
Creating webauthn-api-breakage-prb_shell_run       ... 
Creating webauthn-api-breakage-prb_shell_run       ... done
/bin/bash: line 1: scripts/check_no_api_breakages.sh: Permission denied
126
Build step 'Execute shell' marked build as failure
```

File: docker/docker-compose.2204.main.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ services:
     image: webauthn-swift:22.04-main
     build:
       args:
-        base_image: "swiftlang/swift:nightly-main-focal"
```
Comment (dimitribouniol) [inline docker/docker-compose.2204.main.yaml:9]: @0xTim While debugging, I noticed this was still set to focal, which I _think_ is older? Should we update it here? It Doesn't seem to impact anything yet.

Also, should we add a 5.10 docker compose file? Probably in separate PR?

Comment (yim-lee) [reply docker/docker-compose.2204.main.yaml:9]: > Also, should we add a 5.10 docker compose file?

When ready just @ me in the PR to add 5.10 CI.

Comment (0xTim) [reply docker/docker-compose.2204.main.yaml:9]: @dimitribouniol yeah lets do 5.10 in a new PR

Re jammy vs focal - there's not a huge amount in it but we probably should migrate if you want to do that in the 5.10 PR

**Review by 0xTim — APPROVED** [review]
Thanks @dimitribouniol and @yim-lee !

File: docker/docker-compose.2204.main.yaml (hunk 1 above)
Comment (dimitribouniol) [reply docker/docker-compose.2204.main.yaml:9]: Cool, I’ll tackle this today as well.

Comment (dimitribouniol) [reply docker/docker-compose.2204.main.yaml:9]: 5.10 compose file added here: https://github.com/swift-server/webauthn-swift/pull/55

Seems like focal was just used in the `main` variant — jammy was used everywhere else including the base.

**Pull request merged**